substrate/utils/wasm-builder/src/lib.rs

```diff
@@ -327,6 +333,14 @@ impl CargoCommand {
 		// Check if major and minor are greater or equal than 1.68 or this is a nightly.
 		version.major > 1 || (version.major == 1 && version.minor >= 68) || version.is_nightly
 	}
+
+	/// Returns whether this version of the toolchain supports the `wasm32v1-none` target.
+	fn supports_wasm32v1_none_target(&self) -> bool {
+		self.version.map_or(false, |version| {
+			// Check if major and minor are greater or equal than 1.84.
+			version.major > 1 || (version.major == 1 && version.minor >= 84)
```

Comment:
This check should also include a check of the available targets. So, that we still use `wasm-unknown-unknown` if the new target isn't installed.

Reply:
Do you want it to just run something like `rustup target list --toolchain {toolchain} --installed` every time without any cache? I'm going to try to do it this week, but then I'll be on flights.